tf/tfprocess.py

```diff
@@ -45,26 +48,81 @@ def weight_variable(shape, name=None):
 # Bias weights for layers not followed by BatchNorm
 # We do not regularlize biases, so they are not
 # added to the regularlizer collection
+
+
 def bias_variable(shape, name=None):
     initial = tf.constant(0.0, shape=shape)
     return tf.Variable(initial, name=name)
 
+
 def conv2d(x, W):
     return tf.nn.conv2d(x, W, data_format='NCHW',
                         strides=[1, 1, 1, 1], padding='SAME')
 
+# Restore session from checkpoint. It silently ignore mis-matches
```

Comment:
Is there a reason to add optimistic_restore in this PR?  I prefer to use upgrade.py to avoid accidental upgrades in training.